pkg/ruler/ruler.go

```diff
@@ -593,6 +692,10 @@ func (r *Ruler) syncRules(ctx context.Context, reason string) {
 		return
 	}
 
+	if ctx.Err() != nil {
+		level.Info(r.logger).Log("msg", "context is canceled. not syncing rules")
+		return
+	}
```

Comment:
why we needed to introduce this?

Reply:
When ruler is shutting down/pod terminating, syncs happen. This is because the manager waits for rule groups to stop finishing evaluation. I think it's better to not sync rule groups when ruler is shutting down

Reply:
It just weird that u check it here as this context can be cancelled any place in the code after this line as well and what u trying to prevent can happen anyway?